src/assembly/datasource.ts

```diff
@@ -0,0 +1,45 @@
+import * as host from "./hypermode";
+import { JSON } from "json-as";
+
+
+@json
+class pgInput {
+  query!: string;
+  params!: string[];
+}
+
+
+@json
+class pgResponse {
+  result!: string;
+  error!: string;
+}
+
+export abstract class postgresql {
+  static query(
+    datasourceName: string,
+    query: string,
+    params: string[],
+  ): string {
+    const datasourceType = "postgresql";
+
+    const input: pgInput = {
+      query: query,
+      params: params,
+    };
+
+    const respStr = host.databaseQuery(
+      datasourceName,
+      datasourceType,
+      JSON.stringify<pgInput>(input),
+    );
```

Comment:
Lets pass query and params as two separate variable to the host function.  That will reduce some JSON work for the query, and will make it easier to use the `QueryParameters` object mentioned in the comment in the example.

```suggestion
    const respStr = host.databaseQuery(
      datasourceName,
      datasourceType,
      query,
      params.toJSON()
    );
```